### PR TITLE
MEED-359 Update notes link in favorite drawer if the space name is changed

### PR DESCRIPTION
**Pull request by SaraBoutej**

### CODE REVIEW

File: notes-webapp/src/main/webapp/vue-app/notes-extensions/components/favorites/NotesFavoriteItem.vue [hunk 1]
```diff
@@ -42,7 +42,7 @@ export default {
     this.$notesService.getNoteById(this.id).then(note => {
       const noteSpace = note.wikiOwner.split('/')[2];
       this.noteTitle = note.title;
-      this.noteUrl = `${eXo.env.portal.context}/g/:spaces:${noteSpace}/${noteSpace}/notes/${note.id}`;
+      this.noteUrl = `${eXo.env.portal.context}/g/:spaces:${noteSpace}/${note.url}/notes/${note.id}`;
```
Comment (AzmiTouil) [inline notes-webapp/src/main/webapp/vue-app/notes-extensions/components/favorites/NotesFavoriteItem.vue]:
```suggestion
      this.noteUrl = note.url;
```

**Review by AzmiTouil — CHANGES_REQUESTED** [review]

File: notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java [hunk 2]
```diff
@@ -517,6 +517,9 @@ public Page getNoteById(String id, Identity userIdentity, String source) throws
         throw new IllegalAccessException("User does not have view the note.");
       }
       page.setCanView(true);
+      if (space != null && StringUtils.isEmpty(page.getUrl())) {
+        page.setUrl(space.getUrl());
+      }
```
Comment (AzmiTouil) [inline notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java]:
```suggestion
      page.setUrl(Utils.getPageUrl(page));
```

**Review by AzmiTouil — CHANGES_REQUESTED** [review]

File: notes-webapp/src/main/webapp/vue-app/notes-extensions/components/favorites/NotesFavoriteItem.vue (hunk 1 above)
Comment (SaraBoutej) [reply notes-webapp/src/main/webapp/vue-app/notes-extensions/components/favorites/NotesFavoriteItem.vue]: Done!

File: notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java (hunk 2 above)
Comment (SaraBoutej) [reply notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java]: Done!

File: notes-webapp/src/main/webapp/vue-app/notes-extensions/components/favorites/NotesFavoriteItem.vue [hunk 3]
```diff
@@ -40,9 +40,8 @@ export default {
   }),
   created() {
     this.$notesService.getNoteById(this.id).then(note => {
-      const noteSpace = note.wikiOwner.split('/')[2];
       this.noteTitle = note.title;
-      this.noteUrl = `${eXo.env.portal.context}/g/:spaces:${noteSpace}/${noteSpace}/notes/${note.id}`;
+      this.noteUrl = note.url;
```
Comment (boubaker) [inline notes-webapp/src/main/webapp/vue-app/notes-extensions/components/favorites/NotesFavoriteItem.vue]: may be changing `noteUrl` to a computed value, WDYT ?
and add `this.note = note`

Same for title IMO.

Comment (SaraBoutej) [reply notes-webapp/src/main/webapp/vue-app/notes-extensions/components/favorites/NotesFavoriteItem.vue]: Done!

**Review by AzmiTouil — APPROVED** [review]

**Pull request merged**